src/Resolvers/MapFromResolver.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Spatie\DataTransferObject\Resolvers;
+
+use Spatie\DataTransferObject\Arr;
+use Spatie\DataTransferObject\Attributes\MapFrom;
+use Spatie\DataTransferObject\Descriptors\ClassDescriptor;
+use Spatie\DataTransferObject\Descriptors\PropertyDescriptor;
+
+class MapFromResolver
+{
+    public function __construct(private ClassDescriptor $descriptor)
+    {
+        //
+    }
+
+    public function mapArguments(array $arguments): array
+    {
+        foreach ($this->descriptor->getProperties() as $property) {
+            $arguments = $this->mapArgument($arguments, $property);
+        }
+
+        return $arguments;
+    }
+
+    private function mapArgument(array $arguments, PropertyDescriptor $property): array
+    {
+        $attributes = $property->getReflection()->getAttributes(MapFrom::class);
```

Comment:
Same notes here: let's use spatie/better-types and discuss whether we move the attribute resolving part to the descriptor instead.